internal/peer/node/start.go

```diff
@@ -980,6 +982,18 @@ func registerDiscoveryService(
 	}, support)
 	logger.Info("Discovery service activated")
 	discprotos.RegisterDiscoveryServer(peerServer.Server(), svc)
+	return svc
+}
+
+func registerGateway(peerServer *comm.GRPCServer, discoveryService gateway.DiscoveryService) {
+	if nil == discoveryService {
```

Comment:
We don't use yoda tests in idiomatic go.

Reply:
Do you have any rationale or pointers to go style guides in this area? Testing the other way round reads more nicely but increases the risk of accidentally assigning nil to the variable

Reply:
This isn't C or Java. The condition`if foo = nil` will not even compile in Go. https://play.golang.org/p/B2qhmJMw6Hm

https://github.com/hyperledger/fabric/blob/master/docs/source/style-guides/go-style.rst points to a number of resources that describe what we consider good practices. If you haven't walked through those resources, it's probably worth your time.

Reply:
It may also be worthwhile to look through the list of default `staticcheck` rules.
https://staticcheck.io/docs/checks#ST1017 is the check for yoda conditions.

Reply:
Perhaps that should be added to the list of repository style guides since this idiom doesn't appear to be mentioned in any of the others. Or better still, include staticcheck as part of the build if its rules need to be followed.

Reply:
> include staticcheck as part of the build if its rules need to be followed.

I'd love to see that. Just need to evaluate the ~2000 issues that are currently flagged in master.